spec/controllers/microsoft_controller_spec.rb

```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+describe 'MicrosoftController', type: :request do
+  describe 'GET /.well-known/microsoft-identity-association.json' do
+    it 'successfully retrieves assetlinks.json file' do
+      get '/.well-known/microsoft-identity-association.json'
+      expect(response).to have_http_status(:success)
+      expect(response.body).to include 'applicationId'
+    end
```

Comment:
Can we also ensure that the application id json value = ENV['AZURE_APP_ID']  . we can use `with_modified_env` to  supply a value for the environment variable for this case.